nix/vast/default.nix

```diff
@@ -102,10 +101,12 @@ stdenv.mkDerivation rec {
     "-DVAST_ENABLE_STATIC_EXECUTABLE:BOOL=ON"
     "-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=ON"
   ] ++ lib.optional disableTests "-DVAST_ENABLE_UNIT_TESTS=OFF"
-    # Plugin Section
-    ++ lib.optional (withPlugins != [])
-       "-DVAST_PLUGINS=${lib.concatStringsSep ";" withPlugins}"
-    ++ extraCmakeFlags;
+  # Plugin Section
+  ++ lib.optionals (withPlugins != []) [#Exp: ["broker" "pcap"]
+    "-DVAST_PLUGINS=${lib.concatImapStringsSep ";" (pos: x: "plugins/" + x ) withPlugins}"
```

Comment:
If I understand this correctly you prefix `plugins/` automatically. That would break our closed source CI, which passes additional paths that exist outside of this repository.

Reply:
Fixed
plugins=('"pcap" "broker"') or "${toplevel}" there nothing different.

Reply:
What @dominiklohmann wrote is correct, we need to support plugins from other paths as well.
I think we should create explicit attributes for every plugin and make a `vastWithPlugins` wrapper derivation, but refactorings like that have low prio at the moment.

Reply:
How about remove  "plugins/" to plugins=('"plugins/pcap" "plugins/broker" "<otherpath><plguin>"')?

Reply:
I think we should take a closer look at the problem that this commit solves first. The derivation as written before the change simply added any plugin dependencies as dependencies of VAST itself, which is obviously suboptimal, but the issues goes deeper: One would need to use `vast.overrideAttrs` to add support for plugins with additional dependencies.

A more versatile approach would be to add a `dependencies.nix` file to plugins which need it:
```nix
{ pkgs }:
{
  buildInputs = [ pkgs.libpcap ];
  checkInputs = [ pkgs.tcpdump ];
}
```
We should be able to call this from `vast.nix` and merge all the attributes.

We can't build plugins as separate projects with that approach yet, but it solves at least some of the problems.

Reply:
useful advice, Thanks.